test/unit/uk/gov/hmrc/tariffclassificationfrontend/service/CasesService_ReopenCaseSpec.scala

```diff
@@ -0,0 +1,148 @@
+      eventUpdateFailure(CaseStatus.REFERRED)
+    }
+
+    "not create event on update failure when status is REFERRED" in {
+      eventUpdateFailure(CaseStatus.SUSPENDED)
+    }
+
+    "succeed on event create failure from status REFERRED" in {
+      // Given
+      succeededOnCreateFailure(CaseStatus.REFERRED, CaseStatus.OPEN)
+    }
+
+    "succeed on event create failure from status SUSPENDED" in {
+      // Given
+      succeededOnCreateFailure(CaseStatus.SUSPENDED, CaseStatus.OPEN)
+    }
+
+
+    def updateCaseShould(originalStatus: CaseStatus, updatedStatus: CaseStatus) = {
+      val operator: Operator = Operator("operator-id")
+      val originalCase = aCase.copy(status = originalStatus)
+      val caseUpdated = aCase.copy(status = updatedStatus)
+
+      given(connector.updateCase(any[Case])(any[HeaderCarrier])).willReturn(successful(caseUpdated))
+      given(connector.createEvent(any[Case], any[NewEventRequest])(any[HeaderCarrier])).willReturn(successful(mock[Event]))
+
+      // When Then
+      await(service.reopenCase(originalCase, operator)) shouldBe caseUpdated
+
+      verify(audit).auditCaseReOpen(refEq(originalCase), refEq(caseUpdated), refEq(operator))(any[HeaderCarrier])
+
+      val caseUpdating = theCaseUpdating()
+      caseUpdating.status shouldBe updatedStatus
+
+      val eventCreated = theEventCreatedFor(caseUpdated)
+      eventCreated.userId shouldBe "operator-id"
+      eventCreated.details shouldBe CaseStatusChange(originalStatus, updatedStatus)
+    }
+
+    def eventUpdateFailure(originalStatus: CaseStatus) = {
+      val operator: Operator = Operator("operator-id")
+      val originalCase = aCase.copy(status = originalStatus)
+
+      given(connector.updateCase(any[Case])(any[HeaderCarrier])).willReturn(failed(new RuntimeException()))
+
+      intercept[RuntimeException] {
+        await(service.reopenCase(originalCase, operator))
+      }
+
+      verifyZeroInteractions(audit)
+      verify(connector, never()).createEvent(any[Case], any[NewEventRequest])(any[HeaderCarrier])
+    }
+
+    def succeededOnCreateFailure(originalStatus: CaseStatus, updatedStatus: CaseStatus) = {
+      val operator: Operator = Operator("operator-id")
+      val originalCase = aCase.copy(status = originalStatus)
+      val caseUpdated = aCase.copy(status = updatedStatus)
+
+      given(connector.updateCase(any[Case])(any[HeaderCarrier])).willReturn(successful(caseUpdated))
+      given(connector.createEvent(any[Case], any[NewEventRequest])(any[HeaderCarrier])).willReturn(failed(new RuntimeException()))
+
+      // When Then
+      await(service.reopenCase(originalCase, operator)) shouldBe caseUpdated
+
+      verify(audit).auditCaseReOpen(refEq(originalCase), refEq(caseUpdated), refEq(operator))(any[HeaderCarrier])
+
+      val caseUpdating = theCaseUpdating()
+      caseUpdating.status shouldBe updatedStatus
+    }
+
+  }
+
+  private def theEventCreatedFor(c: Case): NewEventRequest = {
```

Comment:
These two private methods are in few test classes
I would extract them as `protected` methods in a trait to be re-used in all the tests that use them

Reply:
done